jquery.flot.touch.js

```diff
@@ -0,0 +1,134 @@
+
+/* global jQuery */
+
+(function($) {
+    'use strict';
+
+    var options = {
+        pan: {
+            enableTouch: false
+        }
+    };
+
+    function init(plot) {
+        plot.hooks.processOptions.push(initTouchNavigation);
+    }
+
+    function initTouchNavigation(plot, options) {
+        var twoTouches = false;
+
+        function interpretGestures(e) {
+            if (isPinchEvent(e)) {
+                executeAction(e, 'pinch')
+            } else {
+                if (!wasPinchEvent(e)) {
+                    executeAction(e, 'doubleTap')
+                }
+                executeAction(e, 'pan');
+            }
+        }
+
+        function executeAction(e, gesture) {
+            switch (gesture) {
+                case 'pan':
+                    pan[e.type](e);
+                    break;
+                case 'pinch':
+                    pinch[e.type](e)
+                    break;
+                case 'doubleTap':
+                    doubleTap.recenterPlot(e);
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        function bindEvents(plot, eventHolder) {
+            eventHolder[0].addEventListener('touchstart', interpretGestures, false);
+            eventHolder[0].addEventListener('touchmove', interpretGestures, false);
+            eventHolder[0].addEventListener('touchend', interpretGestures, false);
+        }
+
+        function shutdown(plot, eventHolder) {
+            eventHolder[0].removeEventListener('touchstart', interpretGestures);
+            eventHolder[0].removeEventListener('touchmove', interpretGestures);
+            eventHolder[0].removeEventListener('touchend', interpretGestures);
+        }
+
+        var pan = {
+            touchstart: function(e) {
+                preventEventPropagation(e);
+                plot.getPlaceholder()[0].childNodes[2].dispatchEvent(new CustomEvent('panstart', { detail: e }));
```

Comment:
The order of the nodes can change. When bindEvents is called store the eventHolder argument in a member variable and reuse it here.